File: tests/fixtures/Sample.java
package com.example;

/** A service. */
@Service
public class UserService extends Base {
    private final Repo repo;

    public UserService(Repo repo) {
        this.repo = repo;
        init();
        log();
    }

    /** Find a user. */
    public User find(long id) {
        User u = repo.get(id);
        if (u == null) throw new NotFound();
        return u;
    }

    interface Listener {
        void onEvent(String e);
    }

    enum State { ON, OFF }
}

record Point(int x, int y) {}

File: tests/fixtures/Sample.cs
using System;

namespace App.Services
{
    /// <summary>Users.</summary>
    [Serializable]
    public class UserService : IUserService
    {
        private readonly IRepo _repo;

        public UserService(IRepo repo)
        {
            _repo = repo;
            Init();
        }

        public string Name { get; set; }

        public async Task<User> FindAsync(int id)
        {
            var u = await _repo.GetAsync(id);
            if (u == null) throw new NotFoundException();
            return u;
        }
    }

    public interface IUserService
    {
        Task<User> FindAsync(int id);
    }

    public record Point(int X, int Y);
}
